computer/instr.py

```diff
@@ -59,26 +59,19 @@ def process(self, computer):
         computer._b = computer._ram.get_content(int(self._args, 2))
 
         x = int(computer._a, 2) - int(computer._b, 2)
-        x = util._fix_bin(bin(x), binlen=8)
-
-        if '-1' in x:
-            x = '11111111'  # Set to 255
-            computer._carry_flag = 1  # Set carry flag
-            computer._zero_flag = 0   # Prevent zero flag set bug
-
-            # We are sure and we don't want to mess up the flags
-            # so we continue normally and return
-            computer._a = x
-            computer._pc += 1
-
-            return
+        x = util._fix_bin(x, binlen=8)
 
         if len(x) > 8:
             x = x[len(x) - 8:]
             computer._carry_flag = 1
         else:
             computer._carry_flag = 0
 
+        if int(computer._a, 2) == 0:
+            computer._carry_flag = 0
+        else:
+            computer._carry_flag = 1
+

```

Comment:
I think this should test whether `a > b` because `a == 0` is too broad. For example if I were to subtract `0 - 20` this will always set the carry flag to `0` which is incorrect iirc. This also breaks the check for integer overflows above since regardless of whether an overflow happens or not, if `a` was not `0` the flag would just be set to `1` which is probably incorrect (ie. `10+10 = 20` but there is no overflow)`.

I might be wrong here since I haven't worked on this stuff since I first created this repo.

Reply:
The main reason behind this change was based on a line from Ben's video on [Conditional jump instructions](https://youtu.be/Zg1NdPKoosU?t=2255).

> After we subtract 1 here, the carry bit will always be set unless X was 0 before the subtract.

So if we were to view subtraction here as just adding 255 to any value, the only time there wouldn't be an overflow theoretically would be when X == 0. I do agree, however, with the concern about how integer overflows are handled with the commit. As of right now, the code will only detect overflows for subtractions of 1 and it should have that greater flexibility. I'll see what could be done to improve this.